### ISSUE TITLE
Editor-imported `.glb` assets with textures use wrong texture format on Android, resulting in high memory usage

### ISSUE THREAD

**Issue by hybridherbst** [atteneder/glTFast#552]
**Describe the bug**
Seems that when importing a .glb file at editor time, the textures are imported as DXT1 (on Windows at least). When building to Android, it looks like they stay DXT1 and get unpacked as bitmaps at runtime, resulting in giant memory usage. This is a blocker for using glTF as a regular editor-usage format.

Expected would be: when building to Android (and probably other platforms), textures are reimported with the correct platform settings.

**To Reproduce**
1. Import a glb file with embedded textures in the editor
2. Note textures are DXT1 (on Windows at least)
3. Build to Android
4. Note high memory usage (crashes with large files)
5. Import the same thing as glTF with separate textures
6. Build to Android
7. Note expected memory usage (no crash with large files)

**Desktop (please complete the following information):**
 - glTFast version: 4.9.2
 - Unity Editor version 2020.3.42f1
 
additionally (if significant for the bug):

 - Device: Xiaomi Mi 9, Quest 2, Quest Pro

**Additional context**
Regular textures allow per-platform compression settings, not sure how to achieve that for a custom importer:
![image](https://user-images.githubusercontent.com/2693840/210362921-e7fb9f76-67cd-4146-acf6-e20e749fb3f3.png)
``

**Comment by atteneder** [comment]
@hybridherbst Thanks a lot for bringing it up! I'll spend some more thoughts on it soon.

Possibly relevant:

https://docs.unity3d.com/2023.1/Documentation/ScriptReference/AssetImporters.AssetImportContext-selectedBuildTarget.html

**Comment by hybridherbst** [comment]
Thanks!

I'm wondering if in-editor KTX2 imports should maybe always be imported as RGBA32 and then compressed with Unity's built-in compression methods per-platform.

**Comment by atteneder** [comment]
Absolutely! Ideally there would be a way to re-use large portions of the default Texture(2D)ImportSettings inspector.

On top of that (RGBA32 by default) a "do not re-encode" option would make sense in case the KTX already contains the desired format.

**Comment by hybridherbst** [comment]
@atteneder any news on this? :)

**Comment by atteneder** [comment]
I've dabbled with it today. Turns out you can re-encode textures much like a regular textureimporter does via `TextureGenerator.GenerateTexture` (e.g. the 4k textures are automatically downsized to 2k by 2k and encoded to DXT1 on my machine). What's still missing in my PoC is that the texture assignments are broken and there's no UI to actually change texture import settings.

KTX is another beast...it would need the exact same modifications to its importer plus an API that accepts `TextureImporterSettings`/`TextureImporterPlatformSettings`.

I don't have time currently to proceed with it, unfortunately.
